kybra/compiler/kybra_generate/src/canister_method/post_upgrade_method/build.rs

```diff
@@ -13,64 +16,121 @@ use crate::{
     },
     method_utils::params::InternalOrExternal,
     py_ast::PyAst,
+    source_map::SourceMapped,
     Error,
 };
 
 impl PyAst {
-    pub fn build_post_upgrade_method(
-        &self,
-    ) -> Result<(PostUpgradeMethod, TokenStream), Vec<Error>> {
+    pub fn build_post_upgrade_method(&self) -> Result<PostUpgradeMethod, Vec<Error>> {
```

Comment:
I think this could benefit from some [doc?] comments explaining why it is that post_upgrade is taking in the init_params. I know we know at a high-level but in 3 months I might forget the specifics and what we can pull out if we undo this.